web/src/main/java/org/cbioportal/web/StudyViewController.java

```diff
@@ -640,4 +659,151 @@
         return uniqkeyKeys;
     }
     
+    @PreAuthorize("hasPermission(#involvedCancerStudies, 'Collection<CancerStudyId>', 'read')")
+    @RequestMapping(value = "/genomic-data-bin-counts/fetch", method = RequestMethod.POST, consumes = MediaType.APPLICATION_JSON_VALUE, produces = MediaType.APPLICATION_JSON_VALUE)
```

Comment:
Similar comment here as above - can we separate the validation of a filter with the applying of it and also move the calculation of bins to the service layer?